### DISCUSSION TITLE
New action menu for profile switch

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#246] [Q&A] [answered]
Could we have a new Action Menu entry for switching between the profile car and motorcycle and after switching (toggle) automatically forcing a route recalculation? I would place it instead of voice test and move the voice test action to the right into a next 4 action panels area. The next 4 action buttons menu should be reachable by swiping. 
Do you understand what I mean or doesn't it make sense what I write?

The reason is: at the moment I only can drive by car. But normally I mostlyuse Cruiser for motorcycling and this was set to not use motorway. But with the car I wanted to drive on the quickest route to my destination. While driving  it is too dangerous to play around with the sub menues. Therefore a quick change between motorcycle and car profile including new calculation would be wishful to not permanently get routrd  away from the motorway in this case.

**Comment by devemux86** [comment, marked as answer]
@DolomitesTom thanks for the suggestion!

Actions as profile change, routing service change, etc do not belong inside navigation, they can complicate the interface.
They better be done in the normal app interface. After any such changes, we can start navigation with the new selections.

Currently the active profile + router appear in the route menu title. Click on title and the settings appear to change them.

Also the automatic recalculation of routes exists in the long press menu of the routes.
This is to avoid change automatically a route when we simply change profile or router.

So if there is need for a simpler switch of any setting, we can see it, but it does not seem to belong in navigation interface?

**Reply by DolomitesTom** [reply]
Hmm, too much taps while driving to my point of view. In one of the past versions we had the toggle between quick and short routes in the action menu. Why has this been taken out? This would also exactly fit the needs that I mean.

**Reply by 0709wiwiwi** [reply]
While driving ?

**Reply by devemux86** [reply]
The old fastest / shortest route switch was a place holder for symmetry and it is not related to avoidances.

BRouter has detailed options, such switch is not useful. Not until we add BRouter presets for curvy routes.

The navigation interface is for things we do while driving.
Do we switch profiles or select BRouter or "avoid motorways" while driving and not before?

You all know that I am always available to discuss any possible improvements in the interface or options.

If you all want a faster way to switch profiles or any other option, please let's see how it can be dome. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
> Do we switch profiles or select BRouter or "avoid motorways" while driving and not before?

In this case yes and I think it can be useful in some situations. I just wanted to use the quickest way, but it routed me around Auntie's oven, since I couldn't quickly switch to the option "use motorways" without stopping what I didn't want. If others don't find the useful, just archive it.
